src/github.com/getlantern/autoupdate-server/main.go

```diff
@@ -72,6 +72,7 @@ func (u *updateHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			log.Debugf("CheckForUpdate failed with error: %q", err)
 			if err == server.ErrNoUpdateAvailable {
 				u.closeWithStatus(w, http.StatusNoContent)
+				return
```

Comment:
This return was missing, so the execution continued and returned `http.StatusExpectationFailed` instead of `http.StatusNoContent`.